cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -102,6 +106,24 @@
      * discussion in the libraryProvider documentation for complete reasoning.
      */
     protected static ThreadLocal<CqlTerminologyProvider> terminologyProvider = new ThreadLocal<>();
+    
+    protected Map<String, StructType> calculateSparkSchema(List<String> contextNames, ContextDefinitions contextDefinitions) throws Exception{
+        CqlLibraryProvider libProvider = SparkCqlEvaluator.libraryProvider.get();
+        if (libProvider == null) {
+            libProvider = createLibraryProvider();
```

Comment:
I can look at this again. I know I  was having some trouble with creating the thread local since they are static, but some of them were relying on  state from the actual arguments used to  run the program.

Reply:
Now that you say it, that makes sense. We use the input args to the program for both initializers. Probably not worth spending any more cycles worry about it.